### DISCUSSION TITLE
Windows平台报错skia.win32-x64-msvc.node缺少

### DISCUSSION THREAD

**Discussion by GBLKai** [Tsuk1ko/CQ-picfinder-robot#243] [问题求助]
normal.log
`配置已加载`

error.log
```
C:\Users\Kai\Desktop\SetuBot\cq-picsearcher-bot\node_modules\@node-rs\helper\lib\loader.js:1
Error: Can not load bindings, file: C:\Users\Kai\Desktop\SetuBot\cq-picsearcher-bot\node_modules\@napi-rs\canvas-win32-x64-msvc\skia.win32-x64-msvc.node existed but error occurred while require it: The specified module could not be found.
\\?\C:\Users\Kai\Desktop\SetuBot\cq-picsearcher-bot\node_modules\@napi-rs\canvas-win32-x64-msvc\skia.win32-x64-msvc.node 
Installed packages: [canvas-icudtl-win32-x64, canvas-win32-x64-msvc]
    at loadBinding (C:\Users\Kai\Desktop\SetuBot\cq-picsearcher-bot\node_modules\@node-rs\helper\lib\loader.js:50:11)
    at Object.<anonymous> (C:\Users\Kai\Desktop\SetuBot\cq-picsearcher-bot\node_modules\@napi-rs\canvas\index.js:24:5)
    at Generator.next (<anonymous>)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)

```
在提示无法寻找的路径下能发现skia.win32-x64-msvc.node
尝试过安装canvas同样报错
更新之前没有这个问题，更新之后才出现

**Comment by GBLKai** [comment]
操作系统：Windows Server 2019 Datacenter（其他Windows暂时没试）
按[这个](https://github.com/Automattic/node-canvas/wiki/Installation:-Windows)装了canvas
然而安装之后还会报错...

**Reply by Tsuk1ko** [reply]
新版换成 https://github.com/Brooooooklyn/canvas 了

**Comment by GBLKai** [comment]
@Tsuk1ko Node版本v14.17.0

**Reply by Tsuk1ko** [reply]
看起来不是 node 版本的问题，挺迷的

**Comment by Tsuk1ko** [comment]
搞不太懂，我自己的 2012R2 试了一切正常，win 10 也没问题

但也有人跟我反馈有跟你有一样的问题，2008 2012R2 2016 都有问题

不知道是不是 server 系统有什么坑

**Reply by GBLKai** [reply]
会不会是路径的问题 我把程序路径放到桌面了...并且我是中文带UI的server
如果你需要什么系统信息可以继续问我...

**Reply by Tsuk1ko** [reply]
> 会不会是路径的问题 我把程序路径放到桌面了...并且我是中文带UI的server

我也一样，但我没出现这个问题，也和另一个人排查过不是依赖下载之类的问题，所以暂时没有更多头绪……

**Reply by GBLKai** [reply]
> > 会不会是路径的问题
> 
> 我也一样，但我没有出现这个问题，也和另一个人排查过不是依赖下载之类的问题，所以暂时没有更多头绪……

那就在wiki提一嘴？先暂时不管了？我迷惑的地方就是报错和实际出错的地方没什么直接关联...提一下很容易解决这个问题...

**Reply by Tsuk1ko** [reply]
行，等有缘人……

**Comment by GBLKai** [comment]
我再把发现问题的过程写的详细点吧
我本人是小白 只会Google之类的找资料
接到QQ通知的几天之后我尝试更新了一下
所有更新使用的网络活动都是用了代理 代理服务器是日本IP
（QQ并没有GitHub带超链接的文本）
我认为是简单的版本更新就去按平时的流程更新了一下
更新完成后启动一直报错 但是我没有记录那些报错
然后我就把除了config之外都删除了 重新下载了一遍
npm install
报错 缺少文件
又重装了两三次之后
我尝试在文件路径找并且找到了文件
虽然会提示报错 但是程序似乎又在运行
报错的同时 运行日志会刷新新的`配置已加载`
然后自己重启 再报错并且`配置已加载`
（QQ并没有接到上线提醒）

后来Github发出求助后给了一个链接
把链接里的文件放到无法找到文件的目录下
它就能正常运行了

所以其实是它需要手动更新新依赖 但是我以为自动更新的原因？

**Reply by Tsuk1ko** [reply]
> 后来Github发出求助后给了一个链接  
> 把链接里的文件放到无法找到文件的目录下  
> 它就能正常运行了

什么链接

**Comment by benx1n** [comment]
我也遇到了相同的问题
尝试重新获取依赖或手动替换最新的release均报错
但在本地和另一台相同配置的机器上并没有复现

**Reply by benx1n** [reply]
> 我也遇到了相同的问题
> 尝试重新获取依赖或手动替换最新的release均报错
> 但在本地和另一台相同配置的机器上并没有复现

我尝试切换至14.16.1的node后能跑起来，怀疑是node版本的问题

**Comment by Tsuk1ko** [comment]
@GBLKai @benx1n 我有收到反馈说是没装 build tools 的问题（可能跟 vc 依赖有关）

可以试试装 node 的时候最后勾选安装 build tools，或者手动 `npm i -g windows-build-tools`

不过 @napi-rs/canvas 又说是零依赖的，很怪

**Reply by benx1n** [reply]
> @GBLKai @benx1n 我有收到反馈说是没装 build tools 的问题（可能跟 vc 依赖有关）
> 
> 可以试试装 node 的时候最后勾选安装 build tools，或者手动 `npm i -g windows-build-tools`
> 
> 不过 @napi-rs/canvas 又说是零依赖的，很怪

也有可能 我在重新配置node环境时勾选了build tools，如果是需要依赖的话感觉和原生canvas也没差hhh

**Reply by Tsuk1ko** [reply]
至少好处是不用现场编译（
